src/jit/hwintrinsiclistxarch.h

```diff
@@ -212,7 +212,7 @@ HARDWARE_INTRINSIC(SSE2_Or,                                         "Or",
 HARDWARE_INTRINSIC(SSE2_PackSignedSaturate,                         "PackSignedSaturate",                           SSE2,         -1,              16,           2,     {INS_invalid,           INS_invalid,        INS_packsswb,       INS_invalid,        INS_packssdw,       INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid},           HW_Category_SimpleSIMD,             HW_Flag_BaseTypeFromFirstArg)
 HARDWARE_INTRINSIC(SSE2_PackUnsignedSaturate,                       "PackUnsignedSaturate",                         SSE2,         -1,              16,           2,     {INS_invalid,           INS_invalid,        INS_packuswb,       INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid},           HW_Category_SimpleSIMD,             HW_Flag_BaseTypeFromFirstArg)
 HARDWARE_INTRINSIC(SSE2_SetScalarVector128,                         "SetScalarVector128",                           SSE2,         -1,              16,           1,     {INS_invalid,           INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_movsdsse2},         HW_Category_Helper,                 HW_Flag_NoContainment|HW_Flag_MultiIns|HW_Flag_NoRMWSemantics)
-HARDWARE_INTRINSIC(SSE2_SetZeroVector128,                           "SetZeroVector128",                             SSE2,         -1,              16,           0,     {INS_pxor,              INS_pxor,           INS_pxor,           INS_pxor,           INS_pxor,           INS_pxor,           INS_pxor,           INS_pxor,           INS_invalid,        INS_xorpd},             HW_Category_Helper,                 HW_Flag_NoContainment)
+HARDWARE_INTRINSIC(SSE2_SetZeroVector128,                           "SetZeroVector128",                             SSE2,         -1,              16,           0,     {INS_pxor,              INS_pxor,           INS_pxor,           INS_pxor,           INS_pxor,           INS_pxor,           INS_pxor,           INS_pxor,           INS_xorps,          INS_xorpd},             HW_Category_Helper,                 HW_Flag_OneTypeGeneric|HW_Flag_NoContainment)
```

Comment:
This is `generic` and needs the `HW_Flag_OneTypeGeneric` flag.

Since it is generic, and we have not yet determined how we want to expose `SetZeroVector128` (explode it with a modified name, expose it on Sse, or just handle `new Vector128<T>()` properly), I've ensured that `TYP_FLOAT` is handled to unblock PMI and so that we no longer assert for this code path.

Reply:
> how we want to expose SetZeroVector128 

Since we have changed all ISA classes to the inheriti hierarchy, "expose it on Sse" would be the best solution.